src/main/java/org/apache/platypus/server/luceneserver/AnalyzerCreator.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2020 Yelp Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.
+ * See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package org.apache.platypus.server.luceneserver;
+
+import org.apache.lucene.analysis.Analyzer;
+import org.apache.lucene.analysis.custom.CustomAnalyzer;
+import org.apache.lucene.analysis.standard.ClassicAnalyzer;
+import org.apache.lucene.analysis.standard.StandardAnalyzer;
+import org.apache.lucene.util.Version;
+import org.apache.platypus.server.grpc.ConditionalTokenFilter;
+import org.apache.platypus.server.grpc.Field;
+import org.apache.platypus.server.grpc.NameAndParams;
+
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.text.MessageFormat;
+import java.text.ParseException;
+import java.util.HashMap;
+
+public class AnalyzerCreator {
+
+    private static final String LUCENE_ANALYZER_PATH = "org.apache.lucene.analysis.{0}Analyzer";
+
+    static Analyzer getAnalyzer(org.apache.platypus.server.grpc.Analyzer analyzer) {
+        if (!analyzer.getPredefined().isEmpty()) {
+            String predefinedAnalyzer = analyzer.getPredefined();
+
+            if ("standard".equals(predefinedAnalyzer)) {
+                return new StandardAnalyzer();
+            } else if ("classic".equals(predefinedAnalyzer)) {
+                return new ClassicAnalyzer();
+            } else {
+                // Try to dynamically load the analyzer class
+                try {
+                    String className = MessageFormat.format(LUCENE_ANALYZER_PATH, predefinedAnalyzer);
+                    return (Analyzer) AnalyzerCreator.class.getClassLoader().loadClass(className).getDeclaredConstructor().newInstance();
+                } catch (InstantiationException | IllegalAccessException | NoSuchMethodException | ClassNotFoundException | InvocationTargetException e) {
+                    throw new AnalyzerCreationException("Unable to find predefined analyzer: " + predefinedAnalyzer, e);
+                }
+            }
+        } else if (analyzer.hasCustom()) {
+            return getCustomAnalyzer(analyzer.getCustom());
+        } else {
+            return null;
```

Comment:
it seems like the calling code checks for `null` and falls back to using StandardAnalyzer could we potentially just return StandardAnalyzer here followed by a warning log message?

Reply:
Apart from throwing an exception from AnalyzerCreator.getAnalyzer if no analyzer is present, I have also refactored the calling code in RegisterFieldsHandler to hopefully make it more clear.